### PR TITLE
[fix](task) Abort creating replica task if sending RPC failed

### PR DESCRIPTION
**Pull request by w41ter**

### CODE REVIEW

**Comment by w41ter** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/task/AgentBatchTask.java [hunk 1]
```diff
@@ -169,29 +171,34 @@ public void run() {
...
                 } else {
+                    for (AgentTask task : tasks) {
+                        task.failed();
+
+                        // CreateReplicaTask will not trigger a retry in ReportTask.
+                        // Therefore, it needs to be marked as failed here and all
+                        // threads waiting for the result of CreateReplicaTask need
+                        // to be awakened.
+                        if (task instanceof CreateReplicaTask) {
+                            ((CreateReplicaTask) task).countDownToZero(errMsg);
```
Comment (yujun777) [inline fe/fe-core/src/main/java/org/apache/doris/task/AgentBatchTask.java:199]: override CreateReplicaTask::failed()  ? call countDownToZero in its failed

Comment (yujun777) [inline fe/fe-core/src/main/java/org/apache/doris/task/AgentBatchTask.java:199]: 不应该调用  countDownToZero ? 这个函数的意思是让整个latch 不再等待其他的请求，这里应该是latch count - 1.

改成重载?
void CreateReplicaTask::failed() {
    super.failed();
    this.countDownLatch(this.getBackendId(),   this.getSignature())
}

**Comment by w41ter** [comment]
run buildall

**Review by yujun777 — APPROVED** [review]
LGTM

**Review by dataroaring — APPROVED** [review]
LGTM

**Pull request merged**